typescript/src/schema/naming.ts

```diff
@@ -35,20 +35,18 @@ export class Naming {
       throw new Error('Protos provided have different proto packages.');
     }
     const rootPackage = prefix.replace(/\.$/, '');
-
-    // Define the regular expression to match a version component
-    // (e.g. "v1", "v1beta4", etc.).
-    const pattern = /^((?:[a-z0-9_.]+?)\.)?([a-z0-9_]+)(?:\.(v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
-    const match = rootPackage.match(pattern);
-    if (!match) {
+    const match = rootPackage.split('.');
+    if (!match || match.length < 2) {
       throw new Error(`Cannot parse package name ${rootPackage}.`);
     }
-    const [, namespaces, name, version] = match;
-    if (!namespaces) {
-      throw new Error(
-        `Cannot parse package name ${rootPackage}: namespace is not defined.`
-      );
+    const version = match[match.length - 1];
+    // version should follow the pattern of 'v1' or 'v1alpha1'
+    const versionpattern = /^((v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
+    if (!version.match(versionpattern)) {
+      throw new Error(`Cannot parse package name ${rootPackage}.`);
```

Comment:
`Cannot parse package name ${rootPackage}: version does not match ${versionPattern}.` Error messages must explain what went wrong.